### PR TITLE
local_ip: Add interface selection

### LINKED ISSUE #1264: [Feature Request] mac os x - local_ip Interface enX
## Description

There is currently no way to define which interfaces should be used to get the local_ip information.

I have a macbook and the network adapter interface is en4 which is not detected by neofetch.

#### Neofetch version
6.0.0

### LINKED ISSUE #1290: Display multiple [local] IP addresses
## Description

Neofetch only displays one local IP address (presumably the first one it sees with the ip route command?)
On a multi-homed system, is there currently a way to get all network addresses assigned to the system?

#### Neofetch version

6.0.0

### PR DESCRIPTION
**Pull request by iandrewt**
## Description

Adds option `--ip_interface` and array `local_ip_interface` to allow multiple local IPs to be shown

Resolves #1264 and resolves #1290 

## Features

Select as many interfaces as you want
Interfaces with no IP get ignored and error shown when run with `-v`

## Issues

~Only supports macOS for now~ Now supports Linux, BSD, Solaris, AIX and IRIX too (presumably, I was able to test Linux and the other ones used the Linux code originally)

## TODO

- [x] Add support for other OSes

### CODE REVIEW

File: neofetch [hunk 1]
```diff
@@ -382,6 +382,13 @@ public_ip_host="http://ident.me"
 # Flag:    --ip_timeout
 public_ip_timeout=2
 
+# Local IP interface
+#
+# Default: 'default' (interface of default route)
+# Values:  'default', 'en0', 'en1'
+# Flag:    --ip_interface
+local_ip_interface=('default')
```
Comment (dylanaraps) [inline neofetch]: `default` should really be `auto`.

File: neofetch [hunk 2]
```diff
@@ -3364,18 +3371,45 @@ get_battery() {
...
         "Linux" | "BSD" | "Solaris" | "AIX" | "IRIX")
-            local_ip="$(ip route get 1 | awk -F'src' '{print $2; exit}')"
-            local_ip="${local_ip/uid*}"
-            [[ -z "$local_ip" ]] && local_ip="$(ifconfig -a | awk '/broadcast/ {print $2; exit}')"
+            if [[ "${local_ip_interface[0]}" == "default" ]]; then
+                local_ip="$(ip route get 1 | awk -F'src' '{print $2; exit}')"
+                local_ip="${local_ip/uid*}"
+                if [[ -z "$local_ip" ]]; then
+                    local_ip="$(ifconfig -a | awk '/broadcast/ {print $2; exit}')"
+                fi
```
Comment (dylanaraps) [inline neofetch]:
```sh
[[ "$local_ip" ]] || local_ip="$(ifconfig -a | awk bla bla bla)"
```

**Comment by iandrewt** [comment]
Just doing a bit more testing on Linux, its a little broken on my Proxmox box (best place to test something like this).

File: neofetch [hunk 3]
```diff
@@ -3374,14 +3374,14 @@ get_local_ip() {
...
-                [[ "$local_ip" ]] && local_ip="$(ifconfig -a | awk '/broadcast/ {print $2; exit}')"
+                [[ "$local_ip" ]] || local_ip="$(ifconfig -a | awk '/broadcast/ {print $2; exit}')"
             else
                 for interface in "${local_ip_interface[@]}"; do
                     local_ip="$(ip addr show "$interface" | awk '/inet / {print $2; exit}')"
-                    if [[ -n "$local_ip" ]]; then
-                        prin "$interface" "$local_ip"
-                    elif [[ -z "$local_ip" ]]; then
+                    local_ip="${local_ip/\/*}"
+                    [[ "$local_ip" ]] || \
```
Comment (dylanaraps) [inline neofetch]: The `\` is unneeded.

Comment (iandrewt) [reply neofetch]: Huh, TIL

**Comment by iandrewt** [comment]
Should be ready now if you don't have any other suggestions

**Pull request merged**